Mixpanel/MPABTestDesignerConnection.m

```diff
@@ -253,23 +254,61 @@ - (void)webSocket:(MPWebSocket *)webSocket didCloseWithCode:(NSInteger)code reas
     }
 }
 
-- (void)showConnectedView
-{
-    if(!_recordingView) {
+- (void)showConnectedViewWithLoading:(BOOL)isLoading {
+    if (!self.connectivityIndicatorWindow) {
         UIWindow *mainWindow = [[UIApplication sharedApplication] delegate].window;
-        _recordingView = [[UIView alloc] initWithFrame:CGRectMake(0, 0, mainWindow.frame.size.width, 1.0)];
-        _recordingView.backgroundColor = [UIColor colorWithRed:4/255.0f green:180/255.0f blue:4/255.0f alpha:1.0];
-        [mainWindow addSubview:_recordingView];
-        [mainWindow bringSubviewToFront:_recordingView];
+        self.connectivityIndicatorWindow = [[UIWindow alloc] initWithFrame:CGRectMake(0, 0, mainWindow.frame.size.width, 4.f)];
+        self.connectivityIndicatorWindow.backgroundColor = [UIColor clearColor];
+        self.connectivityIndicatorWindow.windowLevel = UIWindowLevelAlert;
+        self.connectivityIndicatorWindow.alpha = 0;
+        self.connectivityIndicatorWindow.hidden = NO;
+        
+        _recordingView = [[UIView alloc] initWithFrame:self.connectivityIndicatorWindow.frame];
+        _recordingView.backgroundColor = [UIColor clearColor];
+        _indeterminateLayer = [CALayer layer];
+        _indeterminateLayer.backgroundColor = [UIColor colorWithRed:1/255.0 green:179/255.0 blue:109/255.0 alpha:1.0].CGColor;
+        _indeterminateLayer.frame = CGRectMake(0, 0, 0, 4.0f);
+        [_recordingView.layer addSublayer:_indeterminateLayer];
+        [self.connectivityIndicatorWindow addSubview:_recordingView];
+        [self.connectivityIndicatorWindow bringSubviewToFront:_recordingView];
+        
+        [UIView animateWithDuration:0.3 animations:^{
+            self.connectivityIndicatorWindow.alpha = 1;
+        }];
     }
+    [self animateConnecting:isLoading];
 }
 
-- (void)hideConnectedView
-{
-    if (_recordingView) {
+- (void)animateConnecting:(BOOL)isLoading {
+    if (isLoading) {
+        CABasicAnimation* myAnimation = [CABasicAnimation animationWithKeyPath:@"bounds.size.width"];
+        myAnimation.duration = 10.f;
+        myAnimation.fromValue = @0;
+        myAnimation.toValue = @(_connectivityIndicatorWindow.bounds.size.width*1.9f);
```

Comment:
Can we pad the operator here `_connectivityIndicatorWindow.bounds.size.width * 1.9f`